mkosi.images/system/mkosi.conf.d/10-arch/mkosi.build.chroot

```diff
@@ -24,12 +24,34 @@ ln --symbolic . "pkg/$ID/src"
 # shellcheck source=/dev/null
 . /etc/makepkg.conf
 
+MKOSI_CFLAGS="-O${OPTIMIZATION:-0} -Wp,-U_FORTIFY_SOURCE"
+if ((LLVM)); then
+    # TODO: Remove -fno-sanitize-function when https://github.com/systemd/systemd/issues/29972 is fixed.
+    MKOSI_CFLAGS="$MKOSI_CFLAGS -shared-libasan -fno-sanitize=function"
+fi
+
+MKOSI_LDFLAGS=""
+if ((LLVM)) && [[ -n "$SANITIZERS" ]]; then
+    MKOSI_LDFLAGS="$MKOSI_LDFLAGS -Wl,-rpath=$(clang --print-file-name="")lib/linux"
+fi
+
+MESON_OPTIONS="$MESON_OPTIONS -D mode=developer -D b_sanitize=${SANITIZERS:-none}"
```

Comment:
I think by analogy with `CFLAGS` it would be nice if `MESON_OPTIONS` supplied externally took precedence over `MESON_OPTIONS` provided by `mkosi`. This way it would be possible to switch to the release mode and so on.